packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/PdfViewer.tsx

```diff
@@ -88,33 +118,74 @@ const PdfViewer: SFC<Props> = ({
     };
   }, [loadedFile, page]);
 
-  const [viewport, canvasInfo] = useMemo(() => {
-    const viewport = loadedPage?.getViewport({ scale });
-    const canvasInfo = viewport ? getCanvasInfo(viewport) : undefined;
-    return [viewport, canvasInfo];
-  }, [loadedPage, scale]);
+  const currentPage = useMemo(() => {
+    const isPageValid = !!loadedPage && loadedPage.pageNumber === page;
+    if (isPageValid) {
```

Comment:
What's the reason for checking if the page is "valid"? We control `loadedPage` and it will either be `null` or a legit PDF page. So seems like all we need to check is if it's non-null, right?

Reply:
The purpose is doing this is to make sure the page is loaded exactly once when user opens a document or changes a page. But I agree that having `loadedPage` and `currentPage` makes the code confusing. So, let me think about refactoring the logic.